### PR TITLE
Prep patches for modifying kargs

### PR DESCRIPTION
**Pull request by jlebon**
```
commit b7e7a01b6d308fe7b85751cefc1f31cd3793db17
Date:   Thu Jun 11 11:41:59 2020 -0400

    install: factor out function to walk BLS configs

    Prep for next patch.

commit 25c71d62f937d52970f85be2294b4f166983d908
Date:   Tue Jun 16 16:38:46 2020 -0400

    install: rename --firstboot-args to --append-firstboot-kargs

    Let's use a more precise name for this switch. First, specify that it
    *appends* to the list of firstboot arguments. Second, use `kargs`
    instead of just `args` to be more clear.

    This is prep for new switches which will use the same naming convention.

commit 70371c47ab76b78f522e4469906e05e1edd05c6d
Date:   Tue Jun 16 17:15:41 2020 -0400

    install: test platform ID restamping bits

    Factor out the closure and add some basic unit tests.
```

### CODE REVIEW

**Comment by jlebon** [comment]
https://github.com/coreos/coreos-ci-lib/commit/853d5fdef3153b2b03c19218a00d516d2b59f853

File: src/install.rs [hunk 1]
```diff
@@ -245,7 +245,30 @@ fn write_platform(mountpoint: &Path, platform: &str) -> Result<()> {
...
 
     eprintln!("Setting platform to {}", platform);
+    edit_bls_entries(mountpoint, |orig_contents: String| {
+        write_platform_in_bls(&orig_contents, platform)
+    })?;
+
+    Ok(())
+}
+
+fn write_platform_in_bls(orig_contents: &str, platform: &str) -> Result<String> {
```
Comment (lucab) [inline src/install.rs:255]: `bls_entry_replace_platform()` maybe? Also, I think you can move portions of the inner comment to a docstring here.

**Comment by lucab** [comment]
Code looks ok, I can stamp this later on if @bgilbert is fine with the new CLI flag arrangement.

File: src/install.rs (hunk 1 above)
Comment (jlebon) [reply src/install.rs]: I went with `bls_entry_write_platform` to match the name of the function `write_platform`. Will update https://github.com/coreos/coreos-installer/pull/268 too to match that naming convention.

**Comment by jlebon** [comment]
Ahhh the rename to `--append-firstboot-kargs` shifts all the help strings to the right and we now go over 80 chars. And there doesn't seem to be a way to tell clap to put the help string on the next line *and* still keep the help string for all other options unindented.

I guess we can keep it as `--firstboot-args`, though I really wanted some consistency with `--append-karg` and `--delete-karg`.

**Comment by jlebon** [comment]
Actually on that subject, couldn't we just "de-emphasize" (if not deprecate) and hide `--firstboot-args` now? The primary use case for this was network configs, and those are now much better covered by `--copy-network`, right?

**Comment by dustymabe** [comment]
> Actually on that subject, couldn't we just "de-emphasize" (if not deprecate) and hide `--firstboot-args` now? The primary use case for this was network configs, and those are now much better covered by `--copy-network`, right?

Maybe we could `de-emphasize`, probably not deprecate at least the underlying code path as `--copy-network` won't apply for the "automation via installer kargs" path.

**Comment by jlebon** [comment]
OK, I updated this now to hide `--firstboot-args`!

**Pull request merged**